File: phase_gate_two/task_three.py
def custom_list_functions(number_list):
    length_counter = 0
    sum_even_position = 0
    sum_odd_position = 0
    multiply_third_position = 1
    sum_of_average = 0
    for num in number_list:
        length_counter += 1
        
        if length_counter % 2 != 0:
            sum_odd_position += num

        if length_counter % 2 == 0:
            sum_even_position += num

        if length_counter % 3 == 0:
            multiply_third_position *= num

        sum_of_average += num
    total_average = sum_of_average/length_counter

    print(f"""
    Length of list: {length_counter}
    Sum of elements in even positions: {sum_even_position}
    Sum of elements in odd positions: {sum_odd_position}
    Multiples elements at every third positions: {multiply_third_position}
    Total average of all elements in the list: {total_average}
    """)
    
 
number_list = []


while True:
    if len(number_list) != 10:
        user_input = int(input("Enter a number ranging from 1 to 50: "))
        if user_input > 50 or user_input < 1 or isinstance(user_input, str):
            print("Invalid input you must enter from 1 to 50!")
            continue
        number_list.append(user_input)
    break


custom_list_functions(number_list)

File: phase_gate_two/java/TaskThree.java
import java.util.Scanner;


public class TaskThree {
    
    public static void main(String[] args) {
        // int[] array = {2, 3, 1, 4, 6, 5, 7, 3};
        // customArrayDetails(array);

        Scanner input = new Scanner(System.in);
        generateCustomUserArray(input);
        
    }


    public static int[] generateCustomUserArray(Scanner input) {
        int[] result = new int[10];
        int counter = 0;
        int userInput;
        while (true) {
            if (counter != 10) {
                userInput = input.nextInt();
                if (userInput < 1 || userInput > 50) {
                    return "Invalid input you must enter from 1 to 50!";
                    continue;
                }
                result[i] = userInput;
            } else {
                return "Array is now up to 10 in length!";
                break;
            }
            counter++;
        }
        customArrayDetails(result);
    }


    public static void customArrayDetails(int[] array) {
        int lengthCounter = 0;
        int sumEvenPosition = 0;
        int sumOddPosition = 0;
        int multiplyThirdPosition = 1;
        int sumOfAverage = 0;

        for (int num : array) {
            lengthCounter ++;
            if (lengthCounter % 2 == 0) sumEvenPosition += num;
            if (lengthCounter % 2 != 0) sumOddPosition += num;
            if (lengthCounter % 3 == 0) multiplyThirdPosition = multiplyThirdPosition * num;
            sumOfAverage += num;
        }
        sumOfAverage = sumOfAverage / lengthCounter;

        System.out.printf("Length of list: %d\n", lengthCounter);
        System.out.printf("Sum of elements in even positions: %d\n", sumEvenPosition);
        System.out.printf("Sum of elements in odd positions: %d\n", sumOddPosition);
        System.out.printf("Multiples elements at every third positions: %d\n", multiplyThirdPosition);
        System.out.printf("Total average of all elements in the list: %d\n", sumOfAverage);
    }
}
